shared/events.js

```diff
@@ -335,8 +355,14 @@ export class MailboxContract extends HashableContract {
   static vuex = MailboxContract.Vuex({
     state: { messages: [] },
     mutations: {
-      PostMessage (state, data) { state.messages.push(data) },
-      PostInvite (state, data) { state.messages.push(data) },
+      PostMessage (state, data) {
+        data.read = false
+        state.messages.push(data)
+      },
+      PostInvite (state, data) {
+        data.read = false
+        state.messages.push(data)
+      },
```

Comment:
Nothing inside of the `data` can be modified, and nothing inside of the `entry` object can be modified. These are immutable objects. So `data.read = false` cannot be done.

We'll need to force this by modifying `Hashable` and `HashableEntry` with accessors throw exceptions when they're written to (will open an issue for this).